src/coreclr/tools/ILVerification/AccessVerificationHelpers.cs

```diff
@@ -22,6 +23,14 @@ internal static bool CanAccess(this TypeDesc currentClass, TypeDesc targetClass)
             if (targetClass.IsParameterizedType)
                 return currentClass.CanAccess(((ParameterizedType)targetClass).ParameterType);
 
+            if (targetClass.IsFunctionPointer)
+            {
+                var signatureBeingPointed = ((FunctionPointerType)targetClass).Signature;
+
+                return currentClass.CanAccess(signatureBeingPointed.ReturnType) &&
+                    signatureBeingPointed._parameters.All(currentClass.CanAccess);
```

Comment:
This should use the public `Signature` surface and not its internal fields.

Reply:
Extracted signature checking in a separate method and removed Linq.